google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -1009,7 +1055,7 @@ private void keepAliveSessions(Instant currTime) {
           logger.log(Level.FINE, "Keeping alive session " + sessionToKeepAlive.getName());
           numSessionsToKeepAlive--;
           sessionToKeepAlive.keepAlive();
-          releaseSession(sessionToKeepAlive);
+          releaseSession(sessionToKeepAlive, Position.FIRST);
```

Comment:
if we re-balancing the pool, we should definitely release the LRU sessions.. Hope this first position does not point to (most recently used) MRU session

Reply:
I'm not sure I understand this comment correctly.

The reasoning behind what is happening here is the following:
1. A session is checked out of the pool by the maintainer, as it has determined that this is a session that needs a keepAlive query.
1. The maintainer executes a `SELECT 1` query on the session, effectively using the session.
1. The maintainer then releases the session back into the pool at the position of the MRU session, as the `SELECT 1` query should be regarded as any other usage of a session.

Should we not consider a `SELECT 1` query executed by the maintainer as a an action that marks the session as most recently used?